syntax = "proto3";

import "common.proto";
import "system_type_base.proto";
package autodrive.workflow.proto;

enum AVP_LPState {
    AVPOff = 0;
    kAbort = 1;
    kStandby = 2;
    kMapLearning = 3;
    kParkinPreActive = 4;
    kCruising = 5;
    kParkInProcess = 6;
    kSuspend = 7;
    kFunctionComplete = 8;
    kMapManageMent = 9;
}

enum ControlCmd {
    kDefault = 0;
    kGetAllMapHeader = 1;
    kGetOneMapInfo = 2;
    kDeleteOneMap = 3;
    kDeleteAllMap = 4;
    kSendTargetParkingSlot = 5;
    kFinishMapLearning = 6;
    kCruisingEnd = 7;
    kReset = 8;
}

message MapLocSystemMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  AVP_LPState avp_lp_state = 3;
  ControlCmd control_cmd = 4;
  uint32 target_parking_slot_id = 5;
  uint32 delete_map_id = 6;
  uint32 select_map_id = 7;
  ApaStateInfo apa_states = 8;
}